course6/toyapp_thomasdhuys/drawlib/primitives.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <iostream>
+
+#include "vertex.h"
+
+namespace Draw {
+
+class RawPrimitive;
+using Primitive = std::unique_ptr<RawPrimitive>;
```

Comment:
This is obfuscation. The name ```Primitive``` sounds like a value, and this is actually a pointer. If you must create alias for the pointer, call it at least ```PrimitvePtr```, but what's wrong with keeping the type explicit?